src/string_utils.cc

```diff
@@ -121,49 +121,63 @@ int str_to_int(StringView str)
 InplaceString<15> to_string(int val)
 {
     InplaceString<15> res;
-    res.m_length = sprintf(res.m_data, "%i", val);
```

Comment:
@lenormf wrote:
> This call cannot overflow the buffer?
> 
> Same remark for all the following hunks,[...]

- The buffer may overflow of the bit size of the `integer` type (implementation dependent) is sufficiently large.
- If the `sprintf` call fails, its negative return value will be recast to the same type as `res.m_length`, which is `unsigned char`. Thus, the failed call will go unchecked, and may cause difficult-to-debug problems further down line (I suppose).

My proposed change takes both of these possibilities into account.

Similar remarks apply to the other hunks.